soem/ethercatcoe.c

```diff
@@ -552,10 +552,9 @@ int ecx_SDOwrite(ecx_contextt *context, uint16 Slave, uint16 Index, uint8 SubInd
 int ecx_RxPDO(ecx_contextt *context, uint16 Slave, uint16 RxPDOnumber, int psize, void *p)
 {
    ec_SDOt *SDOp;
-   int wkc, maxdata;
+   int wkc, maxdata, framedatasize;
```

Comment:
Why is framdatasize promoted to int? Before it was demoted to uint16? uint16 should be enough.

Reply:
Same as previous comments. explicit cast of int psize, demote maxdata to uint16.

   framedatasize = (uint16)psize;
   if (framedatasize > maxdata)
   {
      framedatasize = maxdata;  /*  limit transfer */
   }

Reply:
Give the above code framedatasize must be int. Otherwise the compare if(framedatasize > maxdata) could fail. For example psize = 0x10020. The cast to (uint16) would result in framedatasize = 0x0020. And this is obviously wrong. And I agree CoE transfers>64K are rare, but possible according to the standard.

Reply:
fixed